### PR TITLE
[Az.RecoveryServices.Backup] Added support for AlternateLocation/OriginalLocation restore for managed AzureVMs

### PR DESCRIPTION
**Pull request by hiaga**
## Description

Azure Backup added support for "Create new virtual machine" and "Replace existing virtual machine" experience for Managed VMs in Restore-AzRecoveryServicesBackupItem cmdlet. To perform a VM restore to AlternateLocation use TargetVMName, TargetVNetName, TargetVNetResourceGroup, TargetSubnetName parameters. To perform a restore to a VM in OriginalLocation, do not provide TargetResourceGroupName and RestoreAsUnmanagedDisks parameters, refer examples for more details.

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/RecoveryServices/RecoveryServices.Backup.Providers/Providers/IaasVmPsBackupProvider.cs [hunk 1]
```diff
@@ -474,6 +479,13 @@ public RestAzureNS.AzureOperationResponse TriggerRestore()
                     throw new NotSupportedException(Resources.MSIRestoreNotSupportedForUnmanagedVM);                    
                 }
             }
+            string targetSubscription = ServiceClientAdapter.SubscriptionId;
+             
+            /* if( crossSub != null) // Cross Subscription restore, block unmanaged VMs and other VMs not supported
```
Comment (VeryEarly) [inline src/RecoveryServices/RecoveryServices.Backup.Providers/Providers/IaasVmPsBackupProvider.cs:484]: please remove this part if not needed

**Comment by VeryEarly** [comment]
please suppress breaking change issue

**Comment by hiaga** [comment]
Ack. Will update.

**Comment by hiaga** [comment]
\azp run

**Comment by hiaga** [comment]
@VeryEarly Please review

**Review by VeryEarly — APPROVED** [review]

**Pull request merged**